src/Workspaces/Core/Portable/Diagnostics/Extensions.cs

```diff
@@ -177,6 +191,30 @@ private static string GetAssemblyQualifiedName(Type type)
             return builder.ToImmutable();
         }
 
+        /// <summary>
+        /// Filters out the diagnostics with the specified <paramref name="diagnosticIdsToFilter"/>.
+        /// </summary>
+        public static ImmutableArray<Diagnostic> Filter(this ImmutableArray<Diagnostic> diagnostics, ImmutableArray<string> diagnosticIdsToFilter)
+        {
+            if (diagnosticIdsToFilter.IsEmpty)
+            {
+                return diagnostics;
+            }
+
+            var builder = ArrayBuilder<Diagnostic>.GetInstance(diagnostics.Length);
+            foreach (var diagnostic in diagnostics)
+            {
+                if (diagnosticIdsToFilter.Contains(diagnostic.Id))
+                {
+                    continue;
+                }
+
+                builder.Add(diagnostic);
+            }
+
+            return builder.ToImmutableAndFree();
```

Comment:
Any reason to not go with the easy approach here?

```suggestion
            return diagnostics.RemoveAll(diagnostic => diagnosticIdsToFilter.Contains(diagnostic.Id));
```